src/components/icon/Icon.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import clsx from "clsx";
+import { IconSize } from "./types";
+
+export type IconProps = {
+  /**
+   * The name of the Material Icon to display
+   */
+  name: string;
+
+  /**
+   * Whether the icon should be outlined
+   */
+  outlined?: boolean;
```

Comment:
wdyt instead of a type? If later we want to use more than 2 variants ( rounded, sharp, etc ) it will be cumbersome to use multiple flags for a unique property.